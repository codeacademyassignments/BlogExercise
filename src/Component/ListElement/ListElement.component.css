.list{
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  margin: 2%;
}

.label{
  flex: 1 0 120px;
  max-width: 220px;
  font-size: 20px;
}

.input{
  flex: 1 0 220px;
  height: 47px;
  font-size: 17px;
  border-radius: 9px;
}