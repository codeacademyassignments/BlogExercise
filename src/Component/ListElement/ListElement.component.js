import React from 'react';
import './ListElement.component.css';
import PropTypes from 'prop-types';

const ListElement = ({ inputType, label, placeholder }) => (
  <li className="list">
    <label className="label">{label}</label>
    <input type={inputType} className="input" placeholder={placeholder} />
  </li>
);

ListElement.propTypes = {
  inputType: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
};
export default ListElement;
