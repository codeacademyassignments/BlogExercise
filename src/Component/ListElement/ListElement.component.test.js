import renderer from 'react-test-renderer';
import React from 'react';
import ListElement from './ListElement.component';

describe('ListElement', () => {
  it('ListElement', () => {
    const wrapper = renderer.create(<ListElement label="title" inputType="text" placeholder="enter title" for="title" />);
    expect(wrapper).toMatchSnapshot();
  });
});
