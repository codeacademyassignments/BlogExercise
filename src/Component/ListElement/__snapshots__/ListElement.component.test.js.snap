// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListElement ListElement 1`] = `
<li
  className="list"
>
  <label
    className="label"
  >
    title
  </label>
  <input
    className="input"
    placeholder="enter title"
    type="text"
  />
</li>
`;
